src/isl/util.rs

```diff
@@ -0,0 +1,286 @@
+        use RangeBoundaryValue::*;
+        match (start, end) {
+            (Min, Max) => {
+                return invalid_schema_error("Range boundaries can not be min and max together (i.e. range::[min, max] is not allowed)")
+            }
+            (Min, Value(v, _)) => {
+                Ok(v.to_owned())
+            }
+            (Value(v, _), Max) => {
+                Ok(v.to_owned())
+            }
+            (Value(v1, _), Value(v2, _)) => {
+                match (v1, v2) {
+                    (RangeBoundaryValueType::Integer(_), RangeBoundaryValueType::Integer(_)) => {}
+                    (RangeBoundaryValueType::IntegerNonNegative(_), RangeBoundaryValueType::IntegerNonNegative(_)) => {}
+                    (RangeBoundaryValueType::Decimal(_), RangeBoundaryValueType::Decimal(_)) => {}
+                    (RangeBoundaryValueType::Float(_), RangeBoundaryValueType::Float(_)) => {}
+                    (RangeBoundaryValueType::Timestamp(_), RangeBoundaryValueType::Timestamp(_)) => {},
+                    _ => {
+                        return invalid_schema_error("Both range boundary values must be of same type")
+                    }
+                };
+                // TODO: un-comment below code once the timestamp comparator for ion-rust is implemented
+                // if start > end {
+                //     return invalid_schema_error("Lower range boundary value can not be bigger than upper range boundary")
+                // }
+                Ok(v1.to_owned())
+            }
+            (Max, _) => {
+                return invalid_schema_error("Lower range boundary value must not be max")
+            }
+            (_, Min) => {
+                return invalid_schema_error("Upper range boundary value must not be min")
+            }
+        }
+    }
+}
+
+/// Represents the type of range boundary value
+#[derive(Debug, Clone, PartialEq)]
+pub enum RangeBoundaryValueType {
+    Decimal(Decimal),
+    Float(f64),
+    Integer(i64),
+    IntegerNonNegative(usize),
+    Timestamp(Timestamp),
+}
+
+/// Represents a range boundary value (i.e. min, max or a value in terms of [RangeBoundaryValueType])
+#[derive(Debug, Clone, PartialEq)]
+pub enum RangeBoundaryValue {
+    Max,
+    Min,
+    Value(RangeBoundaryValueType, RangeBoundaryType),
+}
+
+impl RangeBoundaryValue {
+    pub fn int_value(value: i64, range_boundary_type: RangeBoundaryType) -> Self {
+        RangeBoundaryValue::Value(RangeBoundaryValueType::Integer(value), range_boundary_type)
+    }
+    pub fn int_non_negative_value(value: usize, range_boundary_type: RangeBoundaryType) -> Self {
+        RangeBoundaryValue::Value(
+            RangeBoundaryValueType::IntegerNonNegative(value),
+            range_boundary_type,
+        )
+    }
+    pub fn float_value(value: f64, range_boundary_type: RangeBoundaryType) -> Self {
+        RangeBoundaryValue::Value(RangeBoundaryValueType::Float(value), range_boundary_type)
+    }
```

Comment:
Similar TODO thought for these: I think we could have a consolidated `value` method?